[BUG] YAML aliases are treated as strings

**Describe the bug**
YAML aliases are not expanded, causing them to be read literally, as strings, and fail the schema validation.

**To Reproduce**
Steps to reproduce the behavior:

1. Save the example YAML files below.
2. Use the plugin to validate the files WRT the schema below.
4. See error for the version that uses an alias.
5. Notice that the same data without the alias is validated, with or without the anchor.

**Expected behavior**
All files should be found to be valid WRT the schema definition.

**Error messages**
```
[INFO] Starting JSON/YAML validation...
[INFO] Loading schema from: .../integer-list.schema.json
[INFO] Found 3 files to validate
[INFO] Validation complete: 3 files processed, 1 failed
[ERROR] Validation failed for: .../integer-list.yml
[ERROR]   - $.integer-list[3]: string found, integer expected
```

**Configuration**

```xml
<plugin>
    <groupId>com.example</groupId>
    <artifactId>json-yaml-validator-maven-plugin</artifactId>
    <version>1.0.0</version>
    <executions>
      <execution>
        <id>validate-short-yaml</id>
        <phase>validate</phase>
        <goals>
          <goal>validate</goal>
        </goals>
        <configuration>
          <sourceDirectory>.</sourceDirectory>
          <includes>
            <include>integer-list*.yml</include>
          </includes>
          <schemaFile>integer-list.schema.json</schemaFile>
        </configuration>
      </execution>
    </executions>
</plugin>
```

**Environment (please complete the following information):**
- OS: Ubuntu 24.04.3 LTS
- Java Version: 21.0.8
- Maven Version: 3.8.2
- Plugin Version: 1.0.0

**Sample Files**


**JSON Schema:**
```json
{
  "type": "object",
  "properties": {
    "integer-list":
    {
      "type": "array",
      "items": { "type": "integer" }
    }
  }
}
```

**File being validated:**
Validation fails for **integer-list.yml**:
```yml
integer-list: [ &start 1, 3, 5, *start]
```
Validation passes for alias-expanded version **integer-list.alias-expanded.yml**:
```yml
integer-list: [ 1, 3, 5, 1]
```
Validation also passes when placing the anchor without using it in an alias **integer-list.anchor.yml**:
```yml
integer-list: [ &start 1, 3, 5, 1]
```

--- a/src/main/java/com/dataliquid/maven/plugin/schema/validator/JsonYamlValidatorMojo.java
+++ b/src/main/java/com/dataliquid/maven/plugin/schema/validator/JsonYamlValidatorMojo.java
@@ -1,7 +1,8 @@
 package com.dataliquid.maven.plugin.schema.validator;
 
 import java.io.File;
-import java.io.IOException;
+import java.io.FileInputStream;
+import java.io.InputStream;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.ArrayList;
@@ -30,7 +31,7 @@ import com.dataliquid.maven.plugin.schema.validator.utils.AntPatternFileFilter;
 
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import org.yaml.snakeyaml.Yaml;
 import com.networknt.schema.JsonSchema;
 import com.networknt.schema.JsonSchemaFactory;
 import com.networknt.schema.SpecVersion;
@@ -67,8 +68,9 @@ public class JsonYamlValidatorMojo extends AbstractMojo {
     private boolean failOnNoFilesFound;
 
     private final ObjectMapper jsonMapper = new ObjectMapper();
-    private final ObjectMapper yamlMapper = new ObjectMapper(new YAMLFactory());
+    private final ObjectMapper yamlMapper = new ObjectMapper();
 
+    
     @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
         if (skip) {
@@ -253,7 +255,12 @@ public class JsonYamlValidatorMojo extends AbstractMojo {
     private JsonNode readFile(File file) throws IOException {
         String fileName = file.getName().toLowerCase(Locale.ROOT);
         if (fileName.endsWith(".yaml") || fileName.endsWith(".yml")) {
-            return yamlMapper.readTree(file);
+            Yaml snakeYaml = new Yaml();
+            Object loadedYaml;
+            try (InputStream in = new FileInputStream(file)) {
+                loadedYaml = snakeYaml.load(in);
+            }
+            return yamlMapper.valueToTree(loadedYaml);
         } else {
             return jsonMapper.readTree(file);
         }

--- a/src/main/java/com/dataliquid/maven/plugin/schema/validator/JsonYamlValidatorMojo.java
+++ b/src/main/java/com/dataliquid/maven/plugin/schema/validator/JsonYamlValidatorMojo.java
@@ -71,7 +71,6 @@ public class JsonYamlValidatorMojo extends AbstractMojo {
     private final ObjectMapper jsonMapper = new ObjectMapper();
     private final ObjectMapper yamlMapper = new ObjectMapper();
 
-    
     @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
         if (skip) {

--- a/src/main/java/com/dataliquid/maven/plugin/schema/validator/JsonYamlValidatorMojo.java
+++ b/src/main/java/com/dataliquid/maven/plugin/schema/validator/JsonYamlValidatorMojo.java
@@ -1,6 +1,7 @@
 package com.dataliquid.maven.plugin.schema.validator;
 
 import java.io.File;
+import java.io.IOException;
 import java.io.FileInputStream;
 import java.io.InputStream;
 import java.nio.file.Files;

--- a/src/main/java/com/dataliquid/maven/plugin/schema/validator/JsonYamlValidatorMojo.java
+++ b/src/main/java/com/dataliquid/maven/plugin/schema/validator/JsonYamlValidatorMojo.java
@@ -257,7 +257,7 @@ public class JsonYamlValidatorMojo extends AbstractMojo {
         if (fileName.endsWith(".yaml") || fileName.endsWith(".yml")) {
             Yaml snakeYaml = new Yaml();
             Object loadedYaml;
-            try (InputStream in = new FileInputStream(file)) {
+            try (InputStream in = Files.newInputStream(file.toPath())) {
                 loadedYaml = snakeYaml.load(in);
             }
             return yamlMapper.valueToTree(loadedYaml);

--- a/src/test/resources/test-data/v201909/valid/event-with-imports-valid.yaml
+++ b/src/test/resources/test-data/v201909/valid/event-with-imports-valid.yaml
@@ -2,8 +2,8 @@ eventId: EVT-20240620
 title: Virtual DevOps Summit
 description: Online conference for DevOps professionals
 location:
-  latitude: 0
-  longitude: 0
+  latitude: &zero 0
+  longitude: *zero
   timestamp: "2024-02-01T12:00:00Z"
 venue:
   name: Virtual Conference Platform

